File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    
#Solution 1:
#     import re

#     def isPalindrome(self, s: str) -> bool:
#         alphanum = re.sub(r'[^a-zA-Z0-9]',"", s.lower())

#         return alphanum == alphanum[::-1]


# #Solution 2:
    def isPalindrome(self, s: str) -> bool:
        alpha = string.ascii_lowercase
        num = set(str(i) for i in range(10))
        left, right = 0, len(s) - 1
        
        while left < right:
            if s[left].lower() not in alpha and s[left] not in num:
                left += 1
                continue
            if s[right].lower() not in alpha and s[right] not in num:
                right -= 1
                continue
            if s[left].lower() != s[right].lower():
                return False
            left += 1
            right -= 1
        return True
                                            
        
            
            
    

File: 0125-valid-palindrome/0125-valid-palindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    let left = 0;
    let right = s.length - 1
    
    while (left < right) {
        if (!s[left].match(/^[a-zA-Z0-9]/)) {
            left++;
            continue;
        } if (!s[right].match(/^[a-zA-Z0-9]/)) {
            right--;
            continue;
        } if (s[left].toLowerCase() != s[right].toLowerCase()) {
            return false
        }
        left++;
        right--;
    }
    
    return true;
};